app/stores/WalletsStore.js

```diff
@@ -194,6 +220,14 @@ export default class WalletsStore extends Store {
       if (!hasActiveWallet && hasAnyWalletsLoaded) this.active = this.all[0];
       if (this.active) this.goToWalletRoute(this.active.id);
     }
-  }
+  };
+
+  _patchWalletRequestWithWallet = async (wallet) => {
+    await this.walletsRequest.patch(result => {
+      const foundWallet = _.find(result, { id: wallet.id });
+      if (foundWallet) result.pop(foundWallet);
+      result.push(wallet);
+    });
+  };
```

Comment:
Ok that's what i thought … so technically there is no reason for this extra patching logic (it's mostly confusing) since they are exactly the same results.

- if you patch before the results are updated -> fine, that's the reason we do patching because it might take some time until the wallet arrives in other ways.
- if the patch is immediately overwritten, also fine – that's what an optimistic patch is there for (to be replaced) at some later point by the "real" api data.